### PR TITLE
Removed testing of submodules

### PR DESCRIPTION
**Pull request by glatard**
## Description

This removes the testing of sub-datasets, following the discussion in #340

## Related issues
#340

### CODE REVIEW

**Comment by glatard** [comment]
This is ready for review @cmadjar or @emmetaobrien. @mathdugre, if you have 5' to look at this I would sleep better :)

**Comment by mathdugre** [comment]
Hi @glatard, I'll have a look by tomorrow night.

File: tests/functions.py [hunk 1]
```diff
@@ -227,14 +227,10 @@ def authenticate(dataset):
...
-    submodules: Set[str] = get_submodules(dataset)
-    for submodule in submodules:
-        annex_list = git.Repo(os.path.join(dataset, submodule)).git.annex("list")
-        filenames += [
-            os.path.join(submodule, filename)
+    filenames += [
+            filename
             for filename in re.split(r"\n[_X]+\s", annex_list)[1:]
-        ]
+    ]
```
Comment (mathdugre) [inline tests/functions.py:233]: This could be removed without any effects. It duplicates line 229

**Comment by cmadjar** [comment]
@glatard There are conflicts with master now. When you get a chance could you rebase the PR and integrate @mathdugre's suggestion?

Thank you :)

**Comment by glatard** [comment]
I pushed one more commit to my branch but it doesn't seem to be added to this PR, will close and reopen to see if it helps.

**Pull request closed without merging**